cityofaustin/atd-vz-data
# Skip CRIS import record updates if a key column is missing a value

## Associated issues
See https://austininnovation.slack.com/archives/CM9MK950S/p1713190489691989

TLDR: Older records in the person and primary person tables that are missing the `prsn_occpnt_pos_id` are breaking the SQL that we use to query and update records that receive updates from CRIS. This is one of the columns that [we use to check if a person row is unique](https://github.com/cityofaustin/atd-vz-data/blob/5b69bab6f35677c815b412427674da71809b70b7/atd-etl/cris_import/lib/mappings.py#L23-L36).

## Testing
**URL to test:** 
Local

**Steps to test:**
1. Ask me for a zip file that I created that contains csvs with only one crash and its related person, primary person, and unit records that can replicate the original bug 🪄🪱 You can see our current values for the people records in the test csv with the following query (which some of these appear to be duplicates):
```sql
SELECT
	atc.crash_id,
	atc.crash_date,
	atp.prsn_nbr,
	atp.unit_nbr,
	atp.prsn_type_id,
	atp.prsn_occpnt_pos_id
FROM
	atd_txdot_person atp
	LEFT JOIN atd_txdot_crashes atc ON atp.crash_id = atc.crash_id
WHERE
	atc.crash_id = 14003860;
```
2. See readme for local testing instructions
3. Run the import on the zip and you should see the person records that are missing the `prsn_occpnt_pos_id` column value logged as skipped

**Releasing this:**
We should see the [production image of the CR3 import](https://hub.docker.com/layers/atddocker/vz-cris-import/production/images/sha256-9e346d0548dc7a69fbbea4ffc524429da7015596ce4c40eaa2ff7add98ed30c3?context=repo) update after merging this to `master`. [GH action code](https://github.com/cityofaustin/atd-vz-data/blob/5b69bab6f35677c815b412427674da71809b70b7/.github/workflows/build_docker_images.yml#L73-L80)

---
#### Ship list
